### ISSUE TITLE
How to use other otp provider like - msg91

### ISSUE THREAD

**Issue by raghvpip3** [aaronn/django-rest-framework-passwordless#95]
How to use other OTP provider like - msg91

**Comment by jaymes15** [comment]
@raghvpip3 were you able to get it working with Twilio?

**Comment by raghvpip3** [comment]
yes, its working, but I am curious this package will work with other OTP provider or only support for twilio?

**Comment by jaymes15** [comment]
I was only able to get it to work with email

**Comment by jaymes15** [comment]
This the config I used

`PASSWORDLESS_AUTH = {

    'PASSWORDLESS_AUTH_TYPES': ['EMAIL', 'MOBILE'],
    'PASSWORDLESS_EMAIL_NOREPLY_ADDRESS': DEFAULT_FROM_EMAIL,
    'PASSWORDLESS_USER_MOBILE_FIELD_NAME': 'mobile',
    'PASSWORDLESS_MOBILE_NOREPLY_NUMBER': \
        os.environ.get('TWILIO_PHONE_NUMBER'),
    'PASSWORDLESS_EMAIL_TOKEN_HTML_TEMPLATE_NAME': "core/auth.html"
    
}`

**Comment by jaymes15** [comment]
any help would be appreciated

<img width="1792" alt="Screen Shot 2021-09-17 at 6 47 07 AM" src="https://user-images.githubusercontent.com/46538679/133730953-d142e949-581b-4394-9ecb-cec06de4717c.png">

**Comment by jaymes15** [comment]
got it 
it work
<img width="1792" alt="Screen Shot 2021-09-17 at 7 24 03 AM" src="https://user-images.githubusercontent.com/46538679/133734593-276197ed-ab1b-4c34-90de-1f2d7c995325.png">
s

**Comment by dubesar** [comment]
@jaymes15 i am getting the same error, can you share your configs?

**Comment by jaymes15** [comment]
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_USE_TLS = False
EMAIL_HOST = os.environ.get('EMAIL_HOST')
EMAIL_PORT = 587
EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER')
EMAIL_HOST_PASSWORD = os.environ.get('EMAIL_HOST_PASSWORD')
DEFAULT_FROM_EMAIL = EMAIL_HOST_USER


PASSWORDLESS_AUTH = {

    'PASSWORDLESS_AUTH_TYPES': ['EMAIL', 'MOBILE'],
    'PASSWORDLESS_EMAIL_NOREPLY_ADDRESS': DEFAULT_FROM_EMAIL,
    'PASSWORDLESS_USER_MOBILE_FIELD_NAME': 'mobile',
    'PASSWORDLESS_MOBILE_NOREPLY_NUMBER': \
        os.environ.get('TWILIO_PHONE_NUMBER'),
    'PASSWORDLESS_USER_MARK_MOBILE_VERIFIED': True,
    'PASSWORDLESS_USER_MOBILE_VERIFIED_FIELD_NAME': 'mobile_verified',
    'PASSWORDLESS_USER_MARK_EMAIL_VERIFIED': True,
    'PASSWORDLESS_USER_EMAIL_VERIFIED_FIELD_NAME': 'email_verified',
    
    'PASSWORDLESS_EMAIL_TOKEN_HTML_TEMPLATE_NAME': "core/auth.html"
    
}

**Comment by bigfat130397** [comment]
This is possible by mentioning a specific function in configuration, just replace the below value to your function and it will work 

`'PASSWORDLESS_SMS_CALLBACK': 'drfpasswordless.utils.send_sms_with_callback_token'`

for example in my project with app named **testapp** in **views.py** my customized function named **send_sms_code** then my value would be:

`'PASSWORDLESS_SMS_CALLBACK': 'testapp.views.send_sms_code'`

**Comment by jaymes15** [comment]
@bigfat130397 answer is correct 
Tried and it works
